### DISCUSSION TITLE
Better MultiPlatform Support

### DISCUSSION THREAD

**Discussion by pauliesnug** [geode-sdk/sdk#909] [Geode Ideas]
I noticed there was a discussion about _if_ there was plans to make mods universal (https://github.com/geode-sdk/geode/discussions/189). Coming from that, I wonder how a Geode scripting language would look like. Would it's syntax be similar to cpp? 

Also, I wonder if the mod templates and documentation should include more resources and GitHub actions of how to cross-compile mods. If out of the box Geode templates came with cross compilation support, simple mods (which are generally the ones with the least multiplatform support), would automatically have multiplatform support, without the need for any action on the creator's part.

**Comment by matcool** [comment]
on linux, you do not need to change anything about the template mod to get it to cross compile to windows. cross compilation to android obviously works as thats what you're intended to do. 
cross compiling to macos is what has yet to be solved, i failed to even get a simple c++ project to work, but it could be possible!

though all of this does not hide the fact that just getting it to compile != the mod works on said platform. we already offer the multi platform gh action workflow which already builds on all platforms for you.
another reason are the bindings (function addresses and members), which are often missing on some platforms (cough macos)

**Reply by pauliesnug** [reply]
> another reason are the bindings (function addresses and members), which are often missing on some platforms (cough macos)

hmm... i mean, with github actions it's easy to compile a macos mod. i just wonder if there could be more tutorials or just make cross compilation the default... like, maybe some sort of "compatibility checker" which automatically compiles for all platforms available for the codebase?

i noticed in that first thread the issue was too many macos-only mods, now I see too many windows-only mods. i wonder why that changed?
